backend/lib/kubernetes-client/cache/Reflector.js

```diff
@@ -133,29 +121,52 @@ class Reflector {
     return this.lastSyncResourceVersion
   }
 
+  async closeOrTerminateSocket () {
+    if (!this.socket || this.socket.readyState === WebSocket.CLOSED) {
+      logger.debug('Socket of %s has already been closed', this.expectedTypeName)
+      return
+    }
+    const timeout = this.gracePeriod.asMilliseconds()
+    try {
+      logger.debug('Closing socket of %s with timeout %d', this.expectedTypeName, timeout)
+      this.socket.close(4006, 'Gracefully closing websocket')
+      logger.debug('Closing socket of %s has been triggered', this.expectedTypeName)
+      await pEvent(this.socket, 'close', { timeout })
+      logger.debug('Closed socket of %s', this.expectedTypeName)
+    } catch (err) {
+      logger.debug('Terminating socket of %s', this.expectedTypeName)
+      this.socket.terminate()
+      logger.debug('Terminated socket of %s', this.expectedTypeName)
+    }
+  }
+
```

Comment:
I guess we don't need the excessive log here in this function anymore, right?
Also you could move `await pEvent..` on top of `this.socket.close`, just to be sure (as you had it previously)

Reply:
✔️